### PR TITLE
plan: push limit down to table source.

### PR DESCRIPTION
**Pull request by hanfei1991**
push limit down to table scan & index scan. 
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: plan/physical_plans.go [hunk 1]
```diff
@@ -105,10 +105,14 @@ func (p *physicalTableSource) addLimit(l *Limit) {
 }
 
 func (p *physicalTableSource) addTopN(prop *requiredProperty) bool {
+	if len(prop.props) == 0 && prop.limit != nil {
+		p.addLimit(prop.limit)
```
Comment (coocood) [inline plan/physical_plans.go]: If only add limit, the function name should not name as `addTopN`, or add limit should be called out side of `addTopN`.

**Comment by coocood** [comment]
LGTM

File: plan/match_property.go [hunk 2]
```diff
@@ -41,27 +41,30 @@ func (ts *PhysicalTableScan) matchProperty(prop *requiredProperty, infos ...*phy
 		cost = float64(prop.limit.Count+prop.limit.Offset) * netWorkFactor
 	}
 	if len(prop.props) == 0 {
-		p := tryToAddUnionScan(ts.readOnly, ts.conditions, ts)
+		newTs := *ts
```
Comment (zimulala) [inline plan/match_property.go]: s/newTs/newTS

File: plan/match_property.go [hunk 3]
```diff
@@ -41,27 +41,30 @@ func (ts *PhysicalTableScan) matchProperty(prop *requiredProperty, infos ...*phy
...
+		sortedTs.addLimit(prop.limit)
 		p := tryToAddUnionScan(ts.readOnly, ts.conditions, &sortedTs)
 		return enforceProperty(&requiredProperty{limit: prop.limit}, &physicalPlanInfo{
 			p:     p,
 			cost:  cost,
 			count: infos[0].count})
 	}
 	if prop.limit != nil {
-		success := ts.addTopN(prop)
+		sortedTs := *ts
```
Comment (zimulala) [inline plan/match_property.go]: s/sortedTs/sortedTS

**Comment by zimulala** [comment]
LGTM

**Comment by ngaut** [comment]
Tests  can't cover the push down.

**Comment by coocood** [comment]
LGTM

**Comment by hanfei1991** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**